ports/samd/modsamd.c

```diff
@@ -67,8 +73,25 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_1(samd_pininfo_obj, samd_pininfo);
 
 STATIC const mp_rom_map_elem_t samd_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_samd) },
-    { MP_ROM_QSTR(MP_QSTR_Flash),    MP_ROM_PTR(&samd_flash_type) },
     { MP_ROM_QSTR(MP_QSTR_pininfo),  MP_ROM_PTR(&samd_pininfo_obj) },
+    #if MICROPY_HW_MCUFLASH
+    #if defined(MICROPY_HW_QSPIFLASH) || defined(MICROPY_HW_SPIFLASH)
+    #error Only one type of flash must be used
+    #endif
+    { MP_ROM_QSTR(MP_QSTR_Flash), MP_ROM_PTR(&samd_flash_type) },
+    #endif
+    #ifdef MICROPY_HW_QSPIFLASH
+    #if defined(MICROPY_HW_MCUFLASH) || defined(MICROPY_HW_SPIFLASH)
+    #error Only one type of flash must be used
+    #endif
+    { MP_ROM_QSTR(MP_QSTR_Flash), MP_ROM_PTR(&samd_qspiflash_type) },
+    #endif
+    #if defined(MICROPY_HW_SPIFLASH) && defined(MICROPY_HW_SPIFLASH_ID)
+    #if defined(MICROPY_HW_MCUFLASH) || defined(MICROPY_HW_QSPIFLASH)
+    #error Only one type of flash must be used
+    #endif
+    { MP_ROM_QSTR(MP_QSTR_Flash), MP_ROM_PTR(&spiflash_type) },
+    #endif
```

Comment:
This (messy?) if logic could be simplified by either:
1. having a single name for the flash type, `spiflash_type`, instead of `samd_flash_type` and `samd_qspiflash_type`.  Then the linker will give an error if multiple are defined
2. at the top of this file having a `#define SPIFLASH_TYPE <relevant type>` based on which one is configured.  Then the preprocessor will give an error if multiple are defined.

Reply:
I went for Option 2. The code looks much more clean.